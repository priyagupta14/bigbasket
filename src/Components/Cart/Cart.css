.cart-div {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  margin: 2rem;
}
th {
  padding: 1rem;
  background-color: rgb(56, 56, 56);
  color: white;
}
.title {
  display: flex;
  justify-content: space-evenly;
  background-color: rgb(155, 155, 155);
}
.cart-table {
  display: flex;
  flex-direction: column;
  margin: 2rem;
}
.cart-button {
  display: flex;
  /* flex-direction: column; */
  /* flex-item: baseline; */
  justify-content: space-between;
  flex-grow: 1;
}
.cart-button > button {
  box-shadow: 5px 5px 5px grey;
  margin-top: auto;
}
.checkout {
  padding: 5rem;
}
.checkout > button {
  padding: 2em;
  width: 10rem;
  height: 4rem;
  background-color: rgb(245, 211, 117);
  box-shadow: 5px 5px 5px grey;
}
@media screen and (max-width: 24rem) {
  .cart-table {
    display: flex;
  }
}
