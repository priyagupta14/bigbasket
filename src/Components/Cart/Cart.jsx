/* eslint-disable react/prop-types */
import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Table from '../Table/Table';
import './Cart.css';

const Cart = (props) => {
  const { productList, cartCount } = props;

  let totalPrice = Object.keys(productList).map((category) => (
    productList[category].reduce(
      (acc, curVal) => acc + curVal.inCartCount * curVal.price,
      0,
    )));
  totalPrice = totalPrice.reduce(
    (acc, curVal) => acc + curVal,
    0,
  );
  return (
    <div className="cart-div" data-testid="cart-page">
      <h1>
        Your Basket(
        {cartCount}
        item(s))
      </h1>
      <div>
        <Table className="current-cart-table" productList={productList} />
      </div>
      <div className="cart-button">
        <button type="button">
          <Link style={{ textDecoration: 'none', color: 'black' }} to="/">
            CONTINUE SHOPPING
          </Link>
        </button>
        <div className="checkout">
          <div>
            <p>
              TOTAL
            </p>
            <p>
              Rs.
              {totalPrice}
            </p>
          </div>
          <hr />
          <button type="button">
            <Link
              style={{ textDecoration: 'none', color: 'black' }}
              to="/checkout"
            >
              CHECKOUT
            </Link>
          </button>
        </div>
      </div>
    </div>
  );
};
export default Cart;

Cart.propTypes = {
  productList: PropTypes.shape({
    category: PropTypes.arrayOf(
      PropTypes.shape({
        inCartCount: PropTypes.number.isRequired,
        price: PropTypes.number.isRequired,
        name: PropTypes.string.isRequired,
      }),
    ),
  }).isRequired,
  cartCount: PropTypes.number.isRequired,
};
