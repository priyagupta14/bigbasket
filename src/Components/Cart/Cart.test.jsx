/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import {
  fireEvent, render, screen,
} from '@testing-library/react';
import { BrowserRouter, Route } from 'react-router-dom';
import Cart from './Cart';

describe(Cart.name, () => {
  const mockProps = {
    productList: {},
    cartCount: 5,
  };
  it('should match snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        <Cart {...mockProps} />
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });
  it('should display headers', () => {
    render(<BrowserRouter><Cart {...mockProps} /></BrowserRouter>);
    screen.getByText('Your Basket(5item(s))');
    screen.getByText('TOTAL');
    screen.getByText('Rs.0');
  });
  // ask: why it is giving tagname of button as a
  it('should redirect to checkout page', () => {
    render(<BrowserRouter><Cart {...mockProps} /></BrowserRouter>);
    const checkoutPage = screen.getByText('CHECKOUT');
    expect(checkoutPage.tagName).toBe('A');
    fireEvent.click(checkoutPage);
    expect(document.location.href).toBe('http://localhost/checkout');
  });
  it('should redirect to home page', () => {
    render(<BrowserRouter><Route><Cart {...mockProps} /></Route></BrowserRouter>);
    screen.logTestingPlaygroundURL();
    const homePage = screen.getByText('CONTINUE SHOPPING');
    expect(homePage.tagName).toBe('A');
    fireEvent.click(homePage);
    expect(document.location.href).toBe('http://localhost/');
  });
});
