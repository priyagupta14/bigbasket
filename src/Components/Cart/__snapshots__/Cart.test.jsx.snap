// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart should match snapshot 1`] = `
<div>
  <div
    class="cart-div"
  >
    <h1>
      Your Basket(
      5
      item(s))
    </h1>
    <div>
      <table
        border="1"
        class="cart-table"
      >
        <thead>
          <tr
            class="cart-table-header"
          >
            <th>
              ITEM DESCRIPTION
            </th>
            <th>
              UNIT PRICE
            </th>
            <th>
              QUANTITY
            </th>
            <th>
              SUBTOTAL
            </th>
          </tr>
        </thead>
      </table>
    </div>
    <div
      class="cart-button"
    >
      <button
        type="button"
      >
        <a
          href="/"
          style="text-decoration: none; color: black;"
        >
          CONTINUE SHOPPING
        </a>
      </button>
      <div
        class="checkout"
      >
        <div>
          <p>
            TOTAL
          </p>
          <p>
            Rs.
            0
          </p>
        </div>
        <hr />
        <button
          type="button"
        >
          <a
            href="/checkout"
            style="text-decoration: none; color: black;"
          >
            CHECKOUT
          </a>
        </button>
      </div>
    </div>
  </div>
</div>
`;
